Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift

```diff
@@ -235,6 +258,35 @@ extension BrowserViewController: TopToolbarDelegate {
       // Do not allow users to enter URLs with the following schemes.
       // Instead, submit them to the search engine like Chrome-iOS does.
       if !["file"].contains(fixupURL.scheme) {
+        // check text is SNS domain
+        if WalletFeatureFlags.SNSDomainResolverEnabled, fixupURL.domainURL.schemelessAbsoluteDisplayString.endsWithSupportedSNSExtension, let currentStatus = Preferences.Wallet.Web3DomainOption(rawValue: Preferences.Wallet.resolveSNSDomainNames.value) {
+          switch currentStatus {
+          case .ask:
+            // show name service interstitial page
+            showSNSDomainInterstitialPage(originalURL: fixupURL, visitType: visitType)
+            return true
+          case .enabled:
+            Task { @MainActor in
+              let isPrivateMode = PrivateBrowsingManager.shared.isPrivateBrowsing
+              if let rpcService = BraveWallet.JsonRpcServiceFactory.get(privateMode: isPrivateMode) {
+                let (url, status, _) = await rpcService.snsResolveHost(text)
+                if let url = url, status == .success {
+                  // resolved url
+                  finishEditingAndSubmit(url, visitType: visitType)
+                  return true
+                }
+              }
+              // user entered url
+              finishEditingAndSubmit(fixupURL, visitType: visitType)
+              return true
+            }
+          case .disabled:
+            // user entered url
+            finishEditingAndSubmit(fixupURL, visitType: visitType)
+            return true
```

Comment:
```suggestion
          case .disabled:
            break
```

I think it'd be better to fallback to existing logic below here

Reply:
updated in [a4cd1a8](https://github.com/brave/brave-ios/pull/6906/commits/a4cd1a858eb2db471ba0d554b8a31267b8cf01fa)